### PR TITLE
Spelling and formatting cleanup

### PR DESCRIPTION
**Pull request by nealormsbee**
## What does this PR change?
* Adheres to the warnings outlined here: https://goreportcard.com/report/github.com/opencost/opencost

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* It will not.

## Does this PR address any GitHub or Zendesk issues?
* Closes N/A

## How was this PR tested?
* N/A

## Does this PR require changes to documentation?
* No.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
* I cannot label OpenCost PRs

NOTE: I am happy to drop the first commit in this PR if there are aspects of `gofmt -s` which displeases maintainers. The second commit fixes misspellings and should be uncontroversial. Thanks!

### CODE REVIEW

**Review by mattray — APPROVED** [review]
I love that it spellchecked all the comments. Lots of little pedantic fixes, I approve.

**Comment by nealormsbee** [comment]
Thanks @mattray !

Before merging, I'm hoping to get input from an eng maintainer on the `gofmt -s` results. The main difference it makes is to remove explicit type declarations for objects where the type is already implied. If there's an appetite to keep those specific declarations, I'll revert that commit. @michaelmdresser any chance I can bug you to weigh in on that one?

File: pkg/util/allocationfilterutil/v2/parser.go
Comment (michaelmdresser) [inline pkg/util/allocationfilterutil/v2/parser.go]: Please revert these changes, the spacing here is important.

**Review by michaelmdresser — COMMENTED** [review]
I don't have strong feelings about explicit struct names showing up. I'm personally a minor fan of them, but for the sake of being idiomatic we may as well get rid of 'em.

File: pkg/util/allocationfilterutil/v2/parser.go
Comment (michaelmdresser) [reply pkg/util/allocationfilterutil/v2/parser.go]: GitHub didn't grab the whole section, but I meant to grab the whole grammar section in this comment.

Comment (nealormsbee) [reply pkg/util/allocationfilterutil/v2/parser.go]: You got it!

Comment (nealormsbee) [reply pkg/util/allocationfilterutil/v2/parser.go]: Should be good.

**Review by michaelmdresser — APPROVED** [review]

**Comment by nealormsbee** [comment]
Thanks for the approvals y'all. I don't actually have permissions to merge this myself, so go for it if you think it's good.

**Pull request merged**
